### PR TITLE
feat(checkbox): add Checkbox component

### PR DESCRIPTION
**Pull request by lucasaarcoverde**
#### What is the purpose of this pull request?

As the title says.

#### How should this be manually tested?

You can check the `Checkbox` section in the netlify build.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [x] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: src/components/Checkbox/CheckIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+// 🧪 EXPERIMENTAL
+export function CheckIcon() {
+  return (
+    <svg
+      width="16px"
+      height="16px"
+      viewBox="0 0 16 16"
+      fill="currentColor"
+      xmlns="http://www.w3.org/2000/svg"
```
Comment (ThaynanAndrey) [inline src/components/Checkbox/CheckIcon.tsx:10]: Nitpick: You can delete it.

File: src/components/Checkbox/index.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  const ref = useForkRef(innerRef, forwardedRef)
+
+  const { inputProps } = useCheckbox(
+    { isDisabled, ...restProps },
+    { isSelected, setSelected },
+    innerRef
+  )
+
+  const { theme } = useAdminTheme()
+  const currentColor = get(theme, `colors.${color}.default.enable`)
```
Comment (ThaynanAndrey) [inline src/components/Checkbox/index.tsx:46]:
```suggestion
  const currentColor = theme?.colors?.[color].default?.enable
```

Comment (ThaynanAndrey) [reply src/components/Checkbox/index.tsx:46]: Avoid loadash.

File: src/components/Checkbox/index.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+          borderRadius: '15%',
+          borderStyle: 'solid',
+          borderWidth: '2px',
+          justifyContent: 'center',
+          alignItems: 'center',
+          ...focusHollow,
+          ...pickStateTheme({
+            isSelected,
+            isDisabled,
+            color: get(theme, `colors.${color}`),
```
Comment (ThaynanAndrey) [inline src/components/Checkbox/index.tsx:78]: Same, avoid loadash
```suggestion
            color: theme?.colors?.[color]
```

File: src/components/Checkbox/CheckIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+// 🧪 EXPERIMENTAL
+export function CheckIcon() {
+  return (
+    <svg
+      width="16px"
```
Comment (ThaynanAndrey) [inline src/components/Checkbox/CheckIcon.tsx:6]: Nitpick: you don't need use 'px', you can declare with `width={16}` and `height={16}`. It's better when get width and height by param as number.

File: src/components/Checkbox/CheckIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from 'react'
+// 🧪 EXPERIMENTAL
```
[the remark this answers is not in the document]
Comment (lucasaarcoverde) [reply src/components/Checkbox/CheckIcon.tsx:2]: Just a reminder to us that is something that we'll not use it in the future (when we have the icons). We are doing this for now to control what will probably change

File: src/components/Checkbox/CheckIcon.tsx (hunk 4 above)
Comment (lucasaarcoverde) [reply src/components/Checkbox/CheckIcon.tsx:6]: Nice, didn't knew about that! Thanks 🙂

File: src/components/Checkbox/index.tsx (hunk 2 above)
Comment (lucasaarcoverde) [reply src/components/Checkbox/index.tsx:46]: Maybe this is something to be discussed? We are using `lodash` to do this operations in other components too

**Pull request merged**